react-components/src/query/useSearchMappedEquipmentAssetMappings.tsx

```diff
@@ -27,53 +28,92 @@ export type ModelMappingsWithAssets = ModelMappings & {
   assets: Asset[];
 };
 
+export type AssetPage = {
+  assets: Asset[];
+  nextCursor: string | undefined;
+};
+
+export type ModelAssetPage = {
+  modelsAssets: ModelMappingsWithAssets[];
+  nextCursor: string | undefined;
+};
+
 export const useSearchMappedEquipmentAssetMappings = (
   query: string,
   models: AddModelOptions[],
   limit: number = 100,
   userSdk?: CogniteClient
-): UseQueryResult<Asset[]> => {
+): UseInfiniteQueryResult<InfiniteData<AssetPage[]>, Error> => {
   const sdk = useSDK(userSdk);
-  const modelsKey = models.map((model) => [model.modelId, model.revisionId]);
-  const { data: assetMappings, isFetched } = useAllMappedEquipmentAssetMappings(models, sdk);
+  const { data: assetMappingList, isFetched } = useAssetMappedNodesForRevisions(
+    models.map((model) => ({ ...model, type: 'cad' }))
+  );
+  const allAssetMappings = useAllMappedEquipmentAssetMappings(models, sdk);
```

Comment:
What is the different between these two calls? It looks like they fetch the same data?

Reply:
kind of true, but with a catch 😄. `allAssetMappings` will contain only first 1000 mappings which will be displayed/listed which will be quick to display to the user, but when search query key is passed than `assetMappingList` (asset cache data) value will be used.

Reply:
Maybe then is it not better then to change the name of the const to be more understandable based on your description?
Perhaps instead of `const allAssetMappings` then we could have something like
`const initialAssetMappings` if it does make sense, ofc :)